lib/actions/rootActions.js

```diff
@@ -232,11 +234,27 @@ export const setLoading = index => ({
   index
 });
 
-export const selectSavedQuery = (value, index) => ({
-  type: actions.SELECT_SAVED_QUERY,
-  value,
-  index
-});
+export const selectSavedQuery = (value, index) => (dispatch, getState) => {
+  const store = getState();
+  const dashboardInfo = getDashboardInfo(store);
+  const delValue = dashboardInfo.data.items[index].savedQuery;
+  const savedQueriesList = getSavedQueriesList(store);
+  dispatch(changeSavedQueryList(delValue, value));
+  const newStore = getState();
+  const newSavedQueriesList = getSavedQueriesList(newStore);
+  value.length > 0
```

Comment:
In JavaScript 0 is a falsy value so there is no need to check if it's greater than 0.